### DISCUSSION TITLE
error esp32

### DISCUSSION THREAD

**Discussion by denismsd** [renatobo/bonogps#37] [General]
Hello everybody my name is Denis I'm from Brazil, I need help I don't have much knowledge in programming, I'm giving this error

Arduino: 1.8.16 (Windows 8.1), Placa:"ESP32 Dev Module, Disabled, Minimal SPIFFS (1.9MB APP with OTA/190KB SPIFFS), 240MHz (WiFi/BT), QIO, 80MHz, 4MB (32Mb), 921600, Warn"
In file included from C:\Users\DENIS\Documents\Arduino\sketch_nov08b\sketch_nov08b.ino:1:0:
C:\Users\DENIS\Documents\Arduino\libraries\WiFiManager-master/strings_en.h:374:7: error: 'wifi_country_t' does not name a type
 const wifi_country_t WM_COUNTRY_US{"US",1,11,CONFIG_ESP32_PHY_MAX_WIFI_TX_POWER,WIFI_COUNTRY_POLICY_AUTO};

C:\Users\DENIS\Documents\Arduino\libraries\WiFiManager-master/strings_en.h:375:7: error: 'wifi_country_t' does not name a type
 const wifi_country_t WM_COUNTRY_CN{"CN",1,13,CONFIG_ESP32_PHY_MAX_WIFI_TX_POWER,WIFI_COUNTRY_POLICY_AUTO};
C:\Users\DENIS\Documents\Arduino\libraries\WiFiManager-master/strings_en.h:376:7: error: 'wifi_country_t' does not name a type
 const wifi_country_t WM_COUNTRY_JP{"JP",1,14,CONFIG_ESP32_PHY_MAX_WIFI_TX_POWER,WIFI_COUNTRY_POLICY_AUTO};
  ^
Foram encontradas múltiplas bibliotecas para "WiFi.h"
Usado: C:\Users\DENIS\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.2\libraries\WiFi
Não usado: C:\Program Files (x86)\Arduino\libraries\WiFi
exit status 1

**Comment by renatobo** [comment]
Hi Denis!
Have you tried some of the options discussed in this thread: [https://github.com/tzapu/WiFiManager/issues/1183](https://github.com/tzapu/WiFiManager/issues/1183) ?
It looks like the issue is within the WiFiManager library version you are using.

**Reply by denismsd** [reply]
Thank you Renato for the feedback , but it did not work , I also tried using the arduino cloud to see if it was a problem on my pc , but it continued with an error , can I make the recording in esp through PlatformIO ?

I can leave the code without wifi, with setup for esp32 to work with bluetooth and gps in 10hz, without configuration option, I will use it in racecrono

**Reply by renatobo** [reply]
You can most definitely store your preferred "default" configuration and remove libraries you don't need (that's why you find the macros at the beginning of the file - WiFi was not made optional in the current version).

WiFi is required for 

- changing configuration (e.g. switch from one app to another, or change GPS configuration)
- updates via OTA
- interface your GPS to uBlox u-center via TCP/IP

**Comment by denismsd** [comment]
Hi Renato,
thank you so much for all feedback.
I just recorded the code in Esp32, I deleted all the libraries from the arduino, so I put it to verify the code and I started installing the libraries when he asked for it, so it worked.
I'm going to set up the gps and test it on kart,
later I send video of the test.
Thank you very much.

**Reply by renatobo** [reply]
Please remember to setup your base GPS configuration following steps at https://github.com/renatobo/bonogps/tree/main/hardware/GPS#setup-on-your-own-step-by-step
